core/src/core/temporal/utils/helpers.py

```
@@ -0,0 +1,54 @@
+        Input Dictionary or None if there are no dependencies.
+
+    """
+    if not stage_config.depends_on:
+        return None
+
+    input_data = {"dependencies": {}, "records": []}
+
+    for dependency in stage_config.depends_on:
+        if dependency not in stage_data:
+            logger.warning(
+                f"Dependency '{dependency}' not found in stage_data"
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)